helix-core/src/main/java/org/apache/helix/controller/GenericHelixController.java

```diff
@@ -851,7 +865,7 @@ public void onCustomizedStateRootChange(String instanceName, List<String> custom
         }
       }
 
-      _lastSeenCustomizedStateTypes.set(new HashSet<>(customizedStateTypes));
+      lastSeenCustomizedStateTypes = new HashSet<>(customizedStateTypes);
```

Comment:
Are you trying to do the following logic?

// Update the last seen types cache
lastSeenCustomizedStateTypes.clear()
lastSeenCustomizedStateTypes.addAll(customizedStateTypes)

Reply:
Yeah, messed it up.